File: 2024/Day 1/day1.py
num_left: list[int] = []
num_right: list[int] = []
nums: dict[int, int] = {}
total: int = 0

# Part 1
with open("day1_data.txt") as f:
    while line := f.readline():
        left, right = line.split()
        num_left.append(int(left))
        num_right.append(int(right))

for i, j in zip(sorted(num_left), sorted(num_right)):
    total += abs(i-j)

print(total)

# Part 2
total = 0
for i in num_left:
    total += nums.setdefault(i, num_right.count(i))*i

print(total)

File: 2024/Day 1/day1.rs
use std::fs::File;
use std::io::{self, BufRead, BufReader};

fn day1() -> io::Result<i32> {
    let file = File::open("day1_data.txt")?;
    let reader = BufReader::new(file);
    let mut total = 0;

    for line in reader.lines() {
        for s in line.unwrap().split(' ') {
            println!("{}", s);
        }
    }

    /*
    TODO: 
    - split string in each line (with map, maybe?)
    - convert each split entry into numbers 
    - turn it into two separate lists (somehow)
    - sort then iterate through both at once
    - subtract between both numbers and get abs value
    - add all abs values together
    */

    Ok(1)
}

fn main() {
    let _ = day1();
}